src/main/java/de/tum/cit/aet/artemis/assessment/service/ResultService.java

```diff
@@ -623,4 +630,28 @@ public FeedbackAnalysisResponseDTO getFeedbackDetailsOnPage(long exerciseId, Fee
     public long getMaxCountForExercise(long exerciseId) {
         return studentParticipationRepository.findMaxCountForExercise(exerciseId);
     }
+
+    /**
+     * Deletes long feedback texts for the provided list of feedback items.
+     * <br>
+     * This method iterates over the provided list of feedback and checks if each feedback has an associated long feedback text.
+     * If the feedback has a long feedback text and its ID is not null, the method fetches the corresponding {@link LongFeedbackText}
+     * from the repository and deletes it.
+     * <p>
+     * This is useful in cases where long feedback texts need to be removed, such as during feedback cleanup operations.
+     *
+     * @param feedbackList The list of {@link Feedback} objects for which the long feedback texts are to be deleted.
+     *                         Only feedback items that have long feedback texts and a non-null ID will be processed.
+     */
+    public void deleteLongFeedback(List<Feedback> feedbackList) {
+        if (feedbackList == null) {
+            return;
+        }
+        for (Feedback feedback : feedbackList) {
+            if (feedback.getHasLongFeedbackText() && feedback.getId() != null) {
+                Optional<LongFeedbackText> longFeedbackTextOpt = longFeedbackTextRepository.findByFeedbackId(feedback.getId());
+                longFeedbackTextOpt.ifPresent(longFeedbackTextRepository::delete);
+            }
+        }
```

Comment:
```suggestion
        longFeedbackRepository.deleteAllByFeedbackIds(feedbackList.stream().map(DomainObject::getId).toList());
```
Otherwise this might result in many data base calls

Reply:
At least in the diff that is displayed to me right now this does not seem to be addressed - maybe my diff is outdated and doesn't reload properly? 🤔

Reply:
The approach actually does not work. I could not answer why